.github/workflows/publish.yaml

```diff
@@ -35,6 +35,8 @@ jobs:
               run: npm install --legacy-peer-deps
             - name: Build
               run: npm run build
+            - name: Prune modules
+              run: npm prune --production --legacy-peer-deps
```

Comment:
late to party but just for reference, I don't think you technically need this step for how your pipeline is working.
`npm publish` doesn't include stuff in the `node_modules` anyway:
https://docs.npmjs.com/cli/v10/using-npm/developers#keeping-files-out-of-your-package
> Additionally, everything in node_modules is ignored, except for bundled dependencies. npm automatically handles this for you, so don't bother adding node_modules to .npmignore.

So your change below to move the stuff from `dependencies` to `devDependencies` and dropping the packages you don't need overall is the real key to avoid conflicting versions and such when installing this package.  👍 🚮